pkg/controller/cronjob/cronjob_controllerv2_test.go

```diff
@@ -1650,23 +1650,41 @@ func TestControllerV2GetJobsToBeReconciled(t *testing.T) {
 					Namespace:       "foo-ns",
 					Name:            "foo-fooer-owner-ref",
 					Labels:          map[string]string{"key": "different-value"},
-					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef}}},
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob"}}},
 				},
 				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{
 					Namespace:       "foo-ns",
 					Name:            "foo-other-owner-ref",
 					Labels:          map[string]string{"key": "different-value"},
-					OwnerReferences: []metav1.OwnerReference{{Name: "another-cronjob", Controller: &trueRef}}},
+					OwnerReferences: []metav1.OwnerReference{{Name: "another-cronjob", Controller: &trueRef, Kind: "CronJob"}}},
 				},
 			},
 			expected: []*batchv1.Job{{
 				ObjectMeta: metav1.ObjectMeta{
 					Namespace:       "foo-ns",
 					Name:            "foo-fooer-owner-ref",
 					Labels:          map[string]string{"key": "different-value"},
-					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef}}},
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob"}}},
 			}},
 		},
+		{
+			name:    "test getting jobs in namespace with a controller reference and if kind is present, should be CronJob",
+			cronJob: &batchv1.CronJob{ObjectMeta: metav1.ObjectMeta{Namespace: "foo-ns", Name: "fooer"}},
+			jobs: []runtime.Object{
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "foo-ns"}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-diff-owner-ref", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CustomController"}}}},
+				&batchv1.Job{ObjectMeta: metav1.ObjectMeta{Name: "foo-with-owner-ref-cronjob", Namespace: "foo-ns",
+					OwnerReferences: []metav1.OwnerReference{{Name: "fooer", Controller: &trueRef, Kind: "CronJob"}}}},
```

Comment:
can we also test jobs with these scenarios?
- job with the right owner reference but an old UID
- job with the right owner reference but a different group